ds3-autogen-c/src/main/resources/templates/other-templates/ds3_net_c.ftl

```diff
@@ -0,0 +1,570 @@
+                checksum_header = g_strconcat(_get_checksum_type_header(request->checksum_type), checksum_value, NULL);
+                headers = curl_slist_append(headers, checksum_header);
+                g_free(checksum_header);
+            }
+            amz_headers = _canonicalize_amz_headers(request->headers);
+            canonicalized_resource = _canonicalized_resource(request->path, request->query_params);
+            signature = _net_compute_signature(client->log, client->creds, request->verb, canonicalized_resource, date, "", checksum_value, amz_headers);
+
+            g_free(amz_headers);
+            g_free(canonicalized_resource);
+
+            auth_header = g_strconcat("Authorization: AWS ", client->creds->access_id->value, ":", signature, NULL);
+
+            headers = curl_slist_append(headers, auth_header);
+            headers = curl_slist_append(headers, date_header);
+            headers = _append_headers(headers, request->headers);
+
+            curl_easy_setopt(handle, CURLOPT_HTTPHEADER, headers);
+
+            res = curl_easy_perform(handle);
+
+            g_free(date);
+            g_free(date_header);
+            g_free(signature);
+            g_free(auth_header);
+            curl_slist_free_all(headers);
+            curl_easy_cleanup(handle);
+
+            //process the response
+            if (res != CURLE_OK) {
+                char * message = g_strconcat("Request failed: ", curl_easy_strerror(res), NULL);
+                ds3_error* error = ds3_create_error(DS3_ERROR_REQUEST_FAILED, message);
+                g_free(url);
+                g_byte_array_free(response_data.body, TRUE);
+                ds3_str_free(response_data.status_message);
+                ds3_string_multimap_free(response_headers);
+                g_free(message);
+                return error;
+            }
+
+            ds3_log_message(client->log, DS3_DEBUG, "Request completed with status code of: %d", response_data.status_code);
+
+            if (response_data.status_code == 307) {
+                ds3_log_message(client->log, DS3_INFO, "Request encountered a 307 redirect");
+                ds3_str_free(response_data.status_message);
+
+                if (response_data.body != NULL) {
+                    g_byte_array_free(response_data.body, TRUE);
+                }
+                ds3_string_multimap_free(response_headers);
+                retry_count++;
+                ds3_log_message(client->log, DS3_DEBUG, "Retry Attempt: %d | Max Retries: %d", retry_count, client->num_redirects);
+                continue;
+            }
+
+            if (response_data.status_code < 200 || response_data.status_code >= 300) {
+                ds3_error* error = ds3_create_error(DS3_ERROR_BAD_STATUS_CODE, "Got an unexpected status code.");
+                error->error = g_new0(ds3_error_response, 1);
+                error->error->http_error_code = response_data.status_code;
```

Comment:
the members of the ds3_error_response struct changed so I had to modify this are of the existing net code